app/scripts/views/sign_in.js

```diff
@@ -310,4 +325,11 @@ function (Cocktail, p, BaseView, FormView, SignInTemplate, Session,
   );
 
   return View;
+
+  function interTabSignInHandler (event) {
+    if (this.broker.hasCapability('interTabSignIn')) {
+      this.user.setSignedInAccount(event.data)
+        .then(this.navigate.bind(this, 'settings'));
```

Comment:
Wouldn't the OAuth redirect URL be the more logical place to redirect them to though? An OAuth user may not know or care what Firefox Accounts is, so finding a background tab has been redirected to our settings page might seem weird to them.